packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -78,7013 +73,9090 @@
 #endif
 #endif
 
-#ifdef HAVE_ZOLTAN2_OMP
-#include <omp.h>
+// This macro could just be KOKKOS_HAVE_CUDA but preserving it here to
+// facilitate testing this on and off. When ZOLTAN2_MJ_USE_CUDA_KERNEL is
+// defined, the main kernel is run using a parallel_for -> parallel_for loop
+// with atomics, instead of parallel_reduce -> parallel_reduce which gives
+// better performance on host. This occurs because CUDA is more efficient with
+// atomics than host code. I understand ScatterView may be a mechanism to allow
+// a single code pipe-line which runs both reduction or atomic patterns but did
+// not investigate that yet.
+// TODO: Implement ScatterView and get rid of this macro.
+#ifdef KOKKOS_HAVE_CUDA
+#define ZOLTAN2_MJ_USE_CUDA_KERNEL // Atomic Atomic Loops
```

Comment:
This should be a run-time choice.  You might want the atomic updates - based algorithm for OpenMP, depending on the number of threads available at run time.

Reply:
Macro removed and just using KOKKOS_HAVE_CUDA but will discuss with team having it as an option.

Reply:
@MicheldeMessieres Be sure to use `KOKKOS_ENABLE_CUDA`, not `KOKKOS_HAVE_CUDA`.  The latter has long been deprecated and may not exist any more.